src/snowflake/snowpark/_internal/compiler/plan_compiler.py

```diff
@@ -49,6 +49,13 @@ class PlanCompiler:
 
     def __init__(self, plan: SnowflakePlan) -> None:
         self._plan = plan
+        session = plan.session
+        self.cte_optimization_enabled = session.cte_optimization_enabled
```

Comment:
if we are restricting the change to config runtime, this is not necessary anymore, right?

Reply:
we are only putting restriction on `cte_optimization_enabled`. Since for other, we still need to take a snapshot, I'm taking a snapshot for cte param for completeness.

Reply:
since now we are putting protection on all config variables, i think we don't need to do that anymore